example/timer/odp_timer_accuracy.c

```diff
@@ -865,51 +947,48 @@ static void cancel_periodic_timers(test_global_t *test_global)
 		if (odp_timer_periodic_cancel(timer))
 			printf("Failed to cancel periodic timer.\n");
 	}
-
-	while (alloc_timers) {
-		odp_timeout_t tmo;
-		timer_ctx_t *ctx;
-
-		ev = odp_schedule(NULL, ODP_SCHED_WAIT);
-		tmo = odp_timeout_from_event(ev);
-		ctx = odp_timeout_user_ptr(tmo);
-		if (odp_timer_periodic_ack(ctx->timer, ev) != 2)
-			continue;
-		odp_event_free(ev);
-		alloc_timers--;
-	}
 }
 
-static void run_test(test_global_t *test_global)
+static int run_test(void *arg)
 {
-	uint64_t burst, num, num_tmo, next_tmo;
-	uint64_t i, tot_timers;
+	test_global_t *test_global = (test_global_t *)arg;
 	odp_event_t ev;
 	odp_time_t time;
 	uint64_t time_ns, diff_ns;
 	odp_timeout_t tmo;
 	uint64_t tmo_ns;
 	timer_ctx_t *ctx;
-	test_stat_t *stat = &test_global->stat;
 	test_log_t *log = test_global->log;
 	enum mode_e mode = test_global->opt.mode;
+	uint64_t tot_timers = test_global->opt.tot_timers;
+	int tid = odp_thread_id();
+	test_stat_t *stat = &test_global->stat[tid];
 	double period_dbl = test_global->period_dbl;
 	odp_timer_pool_t tp = test_global->timer_pool;
 
 	memset(stat, 0, sizeof(*stat));
 	stat->nsec_before_min = UINT64_MAX;
 	stat->nsec_after_min = UINT64_MAX;
 
-	num      = 0;
-	next_tmo = 1;
-	num_tmo  = test_global->opt.num_warmup + test_global->opt.num;
-	burst    = test_global->opt.burst;
-	tot_timers = test_global->opt.tot_timers;
+	odp_barrier_wait(&test_global->barrier);
+
+	while (1) {
+		ev = odp_schedule(NULL, 10 * ODP_TIME_MSEC_IN_NS);
 
-	for (i = 0; i < tot_timers; i++) {
-		ev = odp_schedule(NULL, ODP_SCHED_WAIT);
+		if (ev == ODP_EVENT_INVALID) {
+			if (mode == MODE_PERIODIC) {
+				if (odp_atomic_load_u64(&test_global->last_events) >=
+				    test_global->opt.alloc_timers)
+					break;
 
-		time = odp_time_local();
+			} else if (odp_atomic_load_u64(&test_global->events) >= tot_timers) {
+				break;
+			}
+
+			continue;
+		}
+
+		time = odp_time_global_strict();
```

Comment:
Likely not a big deal, but this time stamp call could be still moved right after the schedule call. It does not matter if we read timestamp and throw it away few times per second. Maximum accuracy is more important than saving a bit on CPU load.